File: scripts/db/migrations/run-migration.ts
import { createClient } from '@supabase/supabase-js';
import * as fs from 'fs';
import * as path from 'path';
import * as dotenv from 'dotenv';

// Load environment variables from .env.local
dotenv.config({ path: path.join(path.resolve(), '.env.local') });

const supabase = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL!,
  process.env.SUPABASE_SERVICE_ROLE_KEY!
);

async function runMigration() {
  try {
    console.log('🔄 Running case law migration...');
    
    // Read the migration file
    const migrationPath = path.join(path.resolve(), 'scripts/supabase/migration_case_law.sql');
    const migrationSQL = fs.readFileSync(migrationPath, 'utf-8');
    
    // Execute the migration
    const { error } = await supabase.rpc('exec', { sql: migrationSQL });
    
    if (error) {
      // If RPC doesn't work, try direct execution
      console.log('⚠️  RPC method not available, trying direct SQL...');
      
      // Split by semicolons and execute each statement
      const statements = migrationSQL
        .split(';')
        .map(s => s.trim())
        .filter(s => s && !s.startsWith('--'));
      
      for (const statement of statements) {
        console.log(`Executing: ${statement.substring(0, 50)}...`);
        const { error: execError } = await supabase.rpc('exec', { statement });
        
        if (execError && !execError.message.includes('function exec')) {
          console.error(`Error executing statement: ${execError.message}`);
        }
      }
    }
    
    console.log('✅ Migration completed!');
    console.log('⚠️  Note: You may need to run this manually in Supabase SQL Editor if errors occur.');
    
  } catch (error) {
    console.error('Error during migration:', error);
    console.log('\n📋 Manual Migration Steps:');
    console.log('1. Go to https://app.supabase.com/project/rxvuoixenzzxztjlsgms/sql/new');
    console.log('2. Copy and paste the contents of scripts/supabase/migration_case_law.sql');
    console.log('3. Click "Run"');
  }
}

runMigration();

File: scripts/db/migrations/run-migration.py
#!/usr/bin/env python3
"""
Execute SQL migration against Supabase using admin credentials
"""

import os
import sys
import psycopg2
from dotenv import load_dotenv

# Load .env.local
load_dotenv('.env.local')

# Get database credentials from Supabase connection string
# Format: postgres://postgres:[PASSWORD]@[HOST]:5432/postgres
service_role_key = os.getenv('SUPABASE_SERVICE_ROLE_KEY')
supabase_url = os.getenv('NEXT_PUBLIC_SUPABASE_URL')

# Extract host from URL (e.g., https://rxvuoixenzzxztjlsgms.supabase.co -> rxvuoixenzzxztjlsgms.supabase.co)
if supabase_url:
    host = supabase_url.replace('https://', '').replace('http://', '')

# Supabase connection details
db_host = f"{host}"
db_port = "5432"
db_name = "postgres"
db_user = "postgres"
db_password = service_role_key  # Using service role as password is not ideal - would need actual credentials

# Try to get actual DB credentials from .env.local
db_password = os.getenv('SUPABASE_DB_PASSWORD', '')

if not db_password:
    print("❌ Error: SUPABASE_DB_PASSWORD not found in .env.local")
    print("\nTo run migrations, you need database credentials.")
    print("Please run the SQL migration manually:")
    print("1. Go to https://app.supabase.com/project/rxvuoixenzzxztjlsgms/sql/new")
    print("2. Copy and paste: scripts/supabase/migration_case_law.sql")
    print("3. Click 'Run'")
    sys.exit(1)

try:
    # Read migration file
    with open('scripts/supabase/migration_case_law.sql', 'r') as f:
        migration_sql = f.read()
    
    # Connect to database
    print(f"🔄 Connecting to Supabase database at {db_host}...")
    conn = psycopg2.connect(
        host=db_host,
        port=db_port,
        database=db_name,
        user=db_user,
        password=db_password
    )
    
    cursor = conn.cursor()
    
    print("🚀 Running migration...")
    cursor.execute(migration_sql)
    
    conn.commit()
    cursor.close()
    conn.close()
    
    print("✅ Migration completed successfully!")
    
except Exception as e:
    print(f"❌ Error: {e}")
    print("\nPlease run the SQL migration manually:")
    print("1. Go to https://app.supabase.com/project/rxvuoixenzzxztjlsgms/sql/new")
    print("2. Copy and paste: scripts/supabase/migration_case_law.sql")
    print("3. Click 'Run'")
    sys.exit(1)
